code/Test_definitions/DeviceIdentifier_retrieve_identifier.feature

```diff
@@ -0,0 +1,141 @@
+    Then the response status code is 200
+    And the response property "$.imeisv" is IMEISV2
+
+  @DeviceIdentifier_retrieve_identifier103_same_device_different_SIMs
+  Scenario:  retrieve device identifer on the same device but for different SIMs by different telcos. The device identifier MUST be the same.
+    Given they use the base url
+    And the resource is "/retrieve-identifier"
+    And one of the scopes associated with the access token is device-identifier:retrieve-identifier
+    When a device_identifier1 is retrieved on the device with SIMCARD1 in slot1
+    And a device_identifier2 is retrieved on the same device with SIMCARD2 in slot1
+    Then device_identifier1 and device_identifier2 are equal
+
+
+  @DeviceIdentifier_retrieve_identifier102_ipv4Address_success
+  Scenario:  retrieve device identifer for phone number PHONENUMBER2, network connection and access token matches PHONENUMBER2
+    Given they use the base url
+    And the resource is "/retrieve-identifier"
+    And one of the scopes associated with the access token is device-identifier:retrieve-identifier
+    When the HTTPS "POST" request is sent
+    And the request body has the field ipv4Address.publicAddress with a value of PUBLICIPV4ADDRESS
+    And the request body has the field ipv4Address.publicPort with a value of PUBLICPORT
+    And the response header "x-correlator" has same value as the request header "x-correlator"
+    And the response header "Content-Type" is "application/json"
+    And the response body complies with the OAS schema at "/components/schemas/200RetrieveIdentifier"
+    Then the response status code is 200
+    And the response property "$.imeisv" is IMEISV2
+
+  @DeviceIdentifier_retrieve_identifier200_missing_device_information
+  Scenario:  retrieve identifier but no device information in request
+    Given they use the base url
+    And the resource is "/retrieve-identifier"
+    And one of the scopes associated with the access token is device-identifier:retrieve-identifier
+    When the HTTPS "POST" request is sent
+    And the connection the request is sent over originates from a device with PHONENUMBER1
+    And the request body has NO the field phoneNumber nor other device information
+    And the response header "x-correlator" has same value as the request header "x-correlator"
+    And the response header "Content-Type" is "application/json"
+    And the response body complies with the OAS schema at "/components/schemas/ErrorResponse"
+    Then the response status code is 400
+    And the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_ARGUMENT"
+    And the response property "$.message" contains a user friendly text
+
+  @DeviceIdentifier_retrieve_identifier201_missing_scope
+  Scenario:  retrieve device identifier with valid access token but scope device-identifier:retrieve-identifier is missing
+    Given they use the base url
```

Comment:
```suggestion
    Given the correct base url for the API provider is used
```

Reply:
I used the same text as other non-DI test definitions...

Reply:
Well, if I was reviewing other test definitions, I would make the same proposal.

Standards documents should always try to avoid using pronouns due to the risk of ambiguity, particularly on a "standalone" line like this. And there will be many base URLs defined, so we should be clearer about which one to use.

The condition is not particularly important to the test (<i>some</i> base url must be being used, so this condition will always be fulfilled). So we could just remove it. But if we are to keep it, it should be clarified somewhat.